rollup/core/classes/RollupLimits.cls

```diff
@@ -0,0 +1,57 @@
+public without sharing class RollupLimits {
+  @TestVisible
+  private static Integer stubbedQueryRows;
+
+  public class Tester {
+    private final transient RollupControl__mdt control;
+    public Tester(RollupControl__mdt control) {
+      this.control = control;
+    }
+
+    public Boolean hasExceededQueryNumberLimit {
+      get {
+        return this.control?.MaxNumberOfQueries__c < Limits.getQueries();
+      }
+    }
+    public Boolean hasExceededQueryRowLimit {
+      get {
+        return this.getRemainingQueryRows() < 0;
+      }
+    }
+    public Boolean hasExceededHeapSizeLimit {
+      get {
+        return (Limits.getLimitHeapSize() / 2) < Limits.getHeapSize();
```

Comment:
What's the reason for dividing by 2 here? Is that to limit (ha!...?) how much of the heap is consumed by rollup?

Reply:
Just a sentinel value; half the heap seemed like a reasonable approach

Reply:
That makes sense, just wanted to be sure